product-mini/platforms/linux/CMakeLists.txt

```diff
@@ -1,7 +1,9 @@
 # Copyright (C) 2019 Intel Corporation.  All rights reserved.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required (VERSION 2.9)
+cmake_minimum_required (VERSION 3.14)
```

Comment:
A little afraid whether it is friendly to upgrade the cmake version, maybe we can resolve the `pie -fPIE` issue by checking whether the compiler is clang, for example for the CMakeLists.txt of wamrc:
```diff
--- a/wamr-compiler/CMakeLists.txt
+++ b/wamr-compiler/CMakeLists.txt
@@ -237,7 +237,7 @@ endif()
 if (NOT MSVC)
   add_definitions(-D_FORTIFY_SOURCE=2)
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ftrapv")
-  if (NOT WIN32)
+  if (NOT CMAKE_C_COMPILER MATCHES ".*clang.*" AND NOT CMAKE_C_COMPILER_ID MATCHES ".*Clang")
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pie -fPIE")
   endif()
 endif()
```

@loganek, @lum1n0us, @no1wudi What is your opinions?

Reply:
I am ok with the change of [cmake requirement](https://github.com/bytecodealliance/wasm-micro-runtime/issues/1478).

But how about other *CMakeLists.txt*?

Reply:
I'm fine with 3.14 as well, it's already 3 years old and it comes with lots of good features (compared to 2.9) so I think it's worth taking that.

Reply:
rebased